conda-forge.yml

```diff
@@ -1,2 +1,9 @@
+min_py_ver: '35'
 max_py_ver: '37'
-max_r_ver: '35'
\ No newline at end of file
+max_r_ver: '35'
+
+channels:
```

Comment:
I am not sure if this section should be here. I defer to other experts on @conda-forge/staged-recipes. Generally, all dependencies should be in conda-forge.

Reply:
I know that's generally the case, but unfortunately it's currently impossible for `pytorch`, which is why I added that channel (per discussions in gitter)

Reply:
Which gitter conversation was this? Can you link to it? 
We have never allowed other channels in conda-forge.

Reply:
Sure. I don't know how to link to the specific convo, but it's pasted below. Timestamp is Jul 08 2019 17:13.

adbeda:
> question: 
> how can I specify a pinning dependency from a different channel?
> 
> I'm creating a recipe for  snorkel (https://github.com/HazyResearch/snorkel) and this package requires `pytorch==1.1.0` which is doesn't exist in conda-forge channel.  pytorch is available on conda but through its own channel. (i.e. https://anaconda.org/pytorch/pytorch).  
> 
> github issues I read said that its not possible to use different channels at this point but is there a better solution other than attempting to bring pytorch to conda-forge first before working on snorkel?

ascopatz:
> You can use other channels if you set the channels in `conda-forge.yml`: https://conda-forge.org/docs/maintainer/conda_forge_yml.html#channels

> This really isn't encouraged though, because it requires that the user have that channel when they install too

> It is ultimately better to get the deps on conda-forge (or baring that main)

> However, I know what an utter pain pytorch is to build

adbeda:
> ha. thats interesting.  so, what have people done? I mean I can use your suggestions, even when its not encouraged, (to make my life easier) but are there any other acceptable-ish feedstock examples? 

ascopatz:
> You can.  I am not aware of anyone actually doing thing long term
> It is probably OK temporarily

I know it's not recommended/desired, but my impression is that pytorch (and many other GPU-requiring packages) are currently unavailable on conda-forge, despite Herculean efforts on the part of yourself and other maintainers. If there's going to be a hard requirement that any package requiring pytorch and GPUs can't be on conda-forge until those same modules are, there are a significant number of machine learning modules that aren't going to be on conda-forge (until someone with more expertise than a more-or-less pure python data scientist like myself can work on it)

Reply:
> If there's going to be a hard requirement that any package requiring pytorch and GPUs can't be on conda-forge until those same modules are, there are a significant number of machine learning modules that aren't going to be on conda-forge (until someone with more expertise than a more-or-less pure python data scientist like myself can work on it)

It is a hard requirement. We don't have control over other channels and we work closely with defaults which is why defaults is allowed. If you want to lift this requirement, write a CFEP https://github.com/conda-forge/conda-forge-enhancement-proposals and then conda-forge/core can vote on it.

Reply:
That's fine, I'll find an alternative way to distribute my software.

Reply:
@nrweir Can't users be allowed to install Solaris by itself and then pytorch however they want to get it? While building [Snorkel](https://github.com/conda-forge/snorkel-feedstock/blob/master/recipe/meta.yaml#L18-L32), I removed pytorch from a requirement since the pytorch is available on conda-forge  as a cpu variant.

Reply:
Unfortunately pytorch is a dependency and we need it for the `solaris` install.

Reply:
that makes sense. I'll be working to bring `label-maker` and `raster-vision` in the next couple of weeks. let me know if there is anything I can with